packages/driver/src/cy/commands/navigation.ts

```diff
@@ -916,33 +1026,47 @@ export default (Commands, Cypress, cy, state, config) => {
         const existingHash = remote.hash || ''
         const existingAuth = remote.auth || ''
 
-        if (previousDomainVisited && (remote.originPolicy !== existing.originPolicy)) {
+        if (previousUrlVisited && (remote.originPolicy !== existing.originPolicy)) {
           // if we've already visited a new superDomain
           // then die else we'd be in a terrible endless loop
           // we also need to disable retries to prevent the endless loop
           $utils.getTestFromRunnable(state('runnable'))._retries = 0
 
-          return cannotVisitDifferentOrigin(remote.origin, previousDomainVisited, remote, existing, options._log)
-        }
+          const params = { remote, existing, previousUrlVisited, log: options._log }
 
-        const current = $Location.create(win.location.href)
+          return cannotVisitDifferentOrigin(params)
+        }
 
-        // if all that is changing is the hash then we know
-        // the browser won't actually make a new http request
-        // for this, and so we need to resolve onLoad immediately
-        // and bypass the actual visit resolution stuff
-        if (bothUrlsMatchAndOneHasHash(current, remote)) {
-          // https://github.com/cypress-io/cypress/issues/1311
-          if (current.hash === remote.hash) {
-            consoleProps['Note'] = 'Because this visit was to the same hash, the page did not reload and the onBeforeLoad and onLoad callbacks did not fire.'
+        // in a cross origin spec bridge, the window may not have been set yet if nothing has been loaded in the secondary origin,
+        // it's also possible for a new test to start and for a cross-origin failure to occur if the win is set but
+        // the AUT hasn't yet navigated to the secondary origin
+        if (win) {
+          try {
+            const current = $Location.create(win.location.href)
+
+            // if all that is changing is the hash then we know
+            // the browser won't actually make a new http request
+            // for this, and so we need to resolve onLoad immediately
+            // and bypass the actual visit resolution stuff
+            if (bothUrlsMatchAndOneHasHash(current, remote)) {
+              // https://github.com/cypress-io/cypress/issues/1311
+              if (current.hash === remote.hash) {
+                consoleProps['Note'] = 'Because this visit was to the same hash, the page did not reload and the onBeforeLoad and onLoad callbacks did not fire.'
+
+                return onLoad({ runOnLoadCallback: false })
+              }
 
-            return onLoad({ runOnLoadCallback: false })
+              return changeIframeSrc(remote.href, 'hashchange')
+              .then(() => {
+                return onLoad({})
+              })
+            }
+          } catch (e) {
+            // if this is a cross-origin error, skip it
+            if (e.name !== 'SecurityError') {
```

Comment:
Should this be verifying it's a cross origin-error that has already been visited before swallowing the error?

Reply:
The cross origin error is reported as a 'SecurityError', we don't have any more context to go on and the actual error message varies per browser.